tests/experimental/evals/functions/test_classify.py

```diff
@@ -464,6 +455,40 @@ def test_run_relevance_eval_standard_dataframe(
     ]
 
 
+@pytest.mark.respx(base_url="https://api.openai.com/v1/chat/completions", assert_all_called=False)
+def test_classify_tolerance_to_exceptions(
+    classification_dataframe: pd.DataFrame,
+    classification_responses: List[str],
+    classification_template: str,
+    monkeypatch: pytest.MonkeyPatch,
+    respx_mock: respx.mock,
+    capfd,
+):
+    with patch.object(OpenAIModel, "_init_tiktoken", return_value=None):
+        model = OpenAIModel(max_retries=0)
+    queries = classification_dataframe["input"].tolist()
+    for query, response in zip(queries, classification_responses):
+        matcher = M(content__contains=query)
+        # Simulate an error on the second query
+        if query == "What is C++?":
+            response = httpx.Response(500, json={"error": "Internal Server Error"})
+        else:
+            response = httpx.Response(200, json={"choices": [{"message": {"content": response}}]})
+        respx_mock.route(matcher).mock(return_value=response)
+
+    classification_df = llm_classify(
+        dataframe=classification_dataframe,
+        template=classification_template,
+        model=model,
+        rails=["relevant", "irrelevant"],
+    )
+
+    assert classification_df is not None
```

Comment:
could be more specific
```suggestion
    assert classification_df.isna().sum() == 2
```

Reply:
I was foreshadowing the async submission change. Gonna leave it less specific to future proof